### ISSUE TITLE
MW1.27: Exception encountered, of type "LogicException"

### ISSUE THREAD

**Issue by petertambach** [jornane/mwSimpleSamlAuth#31]
When logging in with a new user, the error above is thrown.
with $wgShowExceptionDetails = true; we learn the error comes from includes/user/User.php line 2493.

From 1.27 MediaWiki uses AuthManager for changing authentication data.
In the function setPasswordInternal, line 2488 (called by setInternalPassword) we see we cannot set a password for a user that is not in the database.

WORKAROUND (or fix):
in SimpleSamlAuth.class.php, first add the user to the database, then set password. (Basically swap line 440 & 441 around):
    $user->addToDatabase();
    $user->setInternalPassword( null ); // prevent manual login until reset

**Comment by jornane** [comment]
The opposite is true for MediaWiki <= 1.26. There the code crashes when you add to database.

**Comment by jamesmontalvo3** [comment]
I'm still seeing this issue on MW 1.27. More details to follow.

**Comment by jamesmontalvo3** [comment]
Just looking at the code and not having a lot of background on AuthManager, wondering if this is the issue:

MW 1.27 wants this order as this issue and the corresponding commit have said:
```php
$user->addToDatabase();
$user->setInternalPassword( null ); // prevent manual login until reset
```

To have `addToDatabase()` first, before `setInternalPassword()`. But there are several other `set***()` methods called before `addToDatabase()`:

```php
$changed = false;
if ( isset( $wgSamlRealnameAttr )
	&& isset( $attr[$wgSamlRealnameAttr] )
	&& $attr[$wgSamlRealnameAttr]
	&& $user->getRealName() !== reset( $attr[$wgSamlRealnameAttr] )
) {
	$changed = true;
	$user->setRealName( reset( $attr[$wgSamlRealnameAttr] ) );
}
if ( $attr[$wgSamlMailAttr]
	&& $user->getEmail() !== reset( $attr[$wgSamlMailAttr] )
) {
	$changed = true;
	$user->setEmail( reset( $attr[$wgSamlMailAttr] ) );
	$user->ConfirmEmail();
}
```

Perhaps `setEmail()` and `setRealName()` are the issue?

**Comment by jamesmontalvo3** [comment]
The issue appears to be the missing `global $wgVersion` in that method in the latest release (`v0.7`), as fixed by 8729e270006f1703304d37f7dd3359c84d4425d8 in `master`. @jornane Could you create a new release with this fix? `v0.7.1` perhaps? Thanks!
